client/service/src/task_manager/mod.rs

```diff
@@ -233,11 +252,18 @@ impl TaskManager {
 		prometheus_registry: Option<&Registry>
 	) -> Result<Self, PrometheusError> {
 		let (signal, on_exit) = exit_future::signal();
+
 		// A side-channel for essential tasks to communicate shutdown.
 		let (essential_failed_tx, essential_failed_rx) = tracing_unbounded("mpsc_essential_tasks");
 
 		let metrics = prometheus_registry.map(Metrics::register).transpose()?;
 
+		let (task_notifier, background_tasks) = tracing_unbounded("mpsc_background_tasks");
+		let completion_future = executor.spawn(
+			Box::pin(background_tasks.for_each_concurrent(None, |x| x)),
```

Comment:
`for_each_concurrent` will await on all the JoinHandle future at the same time. It is possible to limit this but I don't think it is necessary. It's actually better for the memory foot print to await them all to not accumulate anything on that stream.

Reply:
added note